### PR TITLE
#5 implement function to search word

### PR DESCRIPTION
**Pull request by hungkv**

### CODE REVIEW

File: app/views/words/_form_search_words.html.erb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<%= form_for :word_list, url: words_path, html: {class: "search-words"}, method: :get, remote: true do |f| %>
```
Comment (chuanhtuan) [inline app/views/words/_form_search_words.html.erb]: em dùng gem RanSack hoặc dùng form_tag cũng được

File: app/views/static_pages/home.html.erb [hunk 2]
```diff
@@ -8,7 +8,7 @@
   <div class="col-md-10">
     <div class="row">
       <div class="col-md-6 text-center">
-        <%= link_to t(:word, scope: :home), lessons_path(), method: :POST, class: "btn btn-primary" %>
+        <%= link_to t(:word, scope: :home), words_path(), method: :GET, class: "btn btn-primary" %>
```
Comment (chuanhtuan) [inline app/views/static_pages/home.html.erb]: <%= link_to t(:word, scope: :home), words_path, class: "btn btn-primary" %>

File: app/models/word.rb [hunk 3]
```diff
@@ -5,4 +5,11 @@ class Word < ActiveRecord::Base
   has_many :lessons, through: :lesson_words
   accepts_nested_attributes_for :answers
   
+  scope :get_all, -> user{}
+  scope :get_learned, -> (user){joins(:lesson_words, :lessons).where("lessons.user_id =?", user.id).uniq
```
Comment (khanhkhanh) [inline app/models/word.rb]: Try changing (user) => user
and where("lessons.user_id =?", user.id) => where(lessons.user_id: user.id)

File: app/models/word.rb [hunk 4]
```diff
@@ -5,4 +5,11 @@ class Word < ActiveRecord::Base
...
+                                                              .where("lesson_words.answer_id is not NULL")}
+  scope :get_not_learned, -> (user){joins(:lesson_words, :lessons).where("lessons.user_id =?", user.id).uniq
```
Comment (khanhkhanh) [inline app/models/word.rb]: The same to the above

File: app/views/words/_form_search_words.html.erb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<%= form_tag words_path, method: :get, remote: true do %>
+  <%= select_tag :category_id, options_for_select(Category.all.collect{|item| [item.title,item.id]}), class: "col-md-4 col-md-offset-3"%>
```
Comment (khanhkhanh) [inline app/views/words/_form_search_words.html.erb]: Category.all => controller

File: app/views/words/_form_search_words.html.erb [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<%= form_tag words_path, method: :get, remote: true do %>
+  <%= select_tag :category_id, options_for_select(Category.all.collect{|item| [item.title,item.id]}), class: "col-md-4 col-md-offset-3"%>  
+  <% Word::STATUS.each_with_index do  |status,index| %>
```
Comment (khanhkhanh) [inline app/views/words/_form_search_words.html.erb]: 2 spaces afer do
and need a space before index

File: app/models/word.rb [hunk 7]
```diff
@@ -5,4 +5,11 @@ class Word < ActiveRecord::Base
   has_many :lessons, through: :lesson_words
   accepts_nested_attributes_for :answers
   
+  scope :get_all, -> user{}
```
Comment (khanhkhanh) [inline app/models/word.rb]: Remove a space before user

File: app/views/words/_word.html.erb [hunk 8]
```diff
@@ -0,0 +1,4 @@
+<tr>
+  <td><%= word.content %></td>
+  <td><%= word.answers.learned.first.content %></td>
+</tr>
```
Comment (khanhkhanh) [inline app/views/words/_word.html.erb]: Break line

File: app/views/words/index.html.erb [hunk 9]
```diff
@@ -0,0 +1,18 @@
+<div class="container">
+  <h2><%= t("words.word_title") %></h2>
```
Comment (khanhkhanh) [inline app/views/words/index.html.erb]: Remove ()

File: app/views/words/index.html.erb [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+  <div class="col-md-12 word-list-filter">
+    <%= render partial: "words/form_search_words" %>
+  </div>
+  <div class="col-md-12">
+    <table class="table table-bordered table-hover">
+      <tr>
+        <th><%= t(:word, scope: :words) %></th>
```
Comment (khanhkhanh) [inline app/views/words/index.html.erb]: Remove ()

File: app/views/words/index.html.erb [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+        <th><%= t(:content, scope: :words) %></th>
```
Comment (khanhkhanh) [inline app/views/words/index.html.erb]: Remove ()

File: app/views/words/index.js.erb [hunk 12]
```diff
@@ -0,0 +1 @@
+$(".show-word").html("<%= escape_javascript(render @words) %>")
```
Comment (khanhkhanh) [inline app/views/words/index.js.erb]: Break line

File: config/routes.rb [hunk 13]
```diff
@@ -6,4 +6,5 @@
   resources :categories, only: :index
   resources :lessons, except: [:delete]
   resources :users, only: [:update, :edit]
+  resources :words, only: :index
```
Comment (khanhkhanh) [inline config/routes.rb]: Remove [] at line 7

File: app/views/words/index.html.erb [hunk 14]
```diff
@@ -0,0 +1,18 @@
...
+        <th><%= t :word, scope: :words %></th>
+        <th><%= t :content, scope: :words %></th>
+      </tr>
+      <tbody class="show-word">
+        <%= render @words %>        
+      </tbody>
+    </table>
+  </div>
+</div>
+
```
Comment (khanhkhanh) [inline app/views/words/index.html.erb]: Remove this line

File: app/views/words/_form_search_words.html.erb [hunk 15]
```diff
@@ -0,0 +1,10 @@
+<%= form_tag words_path, method: :get, remote: true do %>
+  <%= select_tag :category_id, options_for_select(@categories.collect{|item| [item.title,item.id]}), class: "col-md-4 col-md-offset-3"%>  
+  <% Word::STATUS.each_with_index do |status, index| %>
+    <%= radio_button_tag :state, status, checked: index == 0 ? true : false %>
+    <%= label_tag :state, status %>
+  <% end %>
+  <div align="center">
+    <%= submit_tag "Filter", class: "btn btn-primary" %>
```
Comment (khanhkhanh) [inline app/views/words/_form_search_words.html.erb]: i18n

**Pull request merged**